cdc/redo/writer/file.go

```diff
@@ -362,47 +387,25 @@ func (w *Writer) openNew() error {
 	}
 
 	// reset ts used in file name when new file
-	w.commitTS.Store(w.eventCommitTS.Load())
-	w.maxCommitTS.Store(w.eventCommitTS.Load())
-	path := w.filePath() + common.TmpEXT
-	f, err := openTruncFile(path)
-	if err != nil {
-		return cerror.WrapError(cerror.ErrRedoFileOp, errors.Annotate(err, "can't open new redo logfile"))
+	var f *os.File
+	if w.allocator == nil {
+		w.commitTS.Store(w.eventCommitTS.Load())
+		w.maxCommitTS.Store(w.eventCommitTS.Load())
+		path := w.filePath() + common.TmpEXT
+		f, err = openTruncFile(path)
+		if err != nil {
+			return cerror.WrapError(cerror.ErrRedoFileOp, errors.Annotate(err, "can't open new redolog file"))
+		}
+	} else {
+		// if there is a file allocator, we use the pre-created file
+		// supplied by the allocator to boost performance
+		f, err = w.allocator.Open()
+		if err != nil {
+			return cerror.WrapError(cerror.ErrRedoFileOp,
+				errors.Annotate(err, "can't open new redolog file with file allocator"))
+		}
 	}
 	w.file = f
```

Comment:
Should the w.curOff  and  w.oldOff  be inited  here ?

Reply:
These two fields are removed now.